File: backend/src/main/java/com/staccato/category/domain/Role.java
package com.staccato.category.domain;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Getter
public enum Role {

    HOST("host"),
    GUEST("guest");

    private final String role;

    boolean isGuest() {
        return this == Role.GUEST;
    }

    public boolean isHost() {
        return this == Role.HOST;
    }
}

File: android/Staccato_AN/app/src/main/java/com/on/staccato/domain/model/Role.kt
package com.on.staccato.domain.model

enum class Role(val value: String) {
    Host("host"),
    Guest("guest"),
    ;

    companion object {
        fun of(value: String) =
            when (value) {
                "host" -> Host
                "guest" -> Guest
                else -> throw IllegalArgumentException("유효하지 않은 Role 입니다.")
            }
    }
}
